File: Two Pointers/cpp/boj1644.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

vector<int> getPrimeNums(int n)
{
  vector<int> result;
  vector<bool> nums(n + 2, true);

  for (int i = 2; i * i <= n; i++)
    if (nums[i])
      for (int j = i * i; j <= n; j += i)
        nums[j] = false;

  for (int i = 2; i <= n; i++)
    if (nums[i])
      result.push_back(i);

  return result;
}

int main()
{
  cin.tie(NULL);
  ios_base::sync_with_stdio(false);

  int answer = 0;
  int n;
  cin >> n;
  vector<int> primeNums = getPrimeNums(n);

  int lt = 0;
  int sum = 0;
  for (int rt = 0; rt < primeNums.size(); rt++)
  {
    sum += primeNums[rt];
    while (n < sum)
      sum -= primeNums[lt++];

    answer += sum == n ? 1 : 0;
  }

  cout << answer;
}

File: Two Pointers/boj1644.js
function getPrimeNums(n) {
  let result = [];
  let nums = Array(n + 2).fill(true);

  for (let i = 2; i * i <= n; i++) {
    if (nums[i]) {
      for (let j = i * i; j <= n; j += i) {
        nums[j] = false;
      }
    }
  }

  for (let i = 2; i <= n; i++) {
    if (nums[i]) result.push(i);
  }
  return result;
}

function solution(n) {
  let answer = 0;
  let pirmeNums = getPrimeNums(n);

  let left = 0;
  let sum = 0;

  for (let right = 0; right < pirmeNums.length; right++) {
    sum += pirmeNums[right];

    while (n < sum) {
      sum -= pirmeNums[left++];
    }
    answer += sum === n ? 1 : 0;
  }

  return answer;
}
